### PR TITLE
refactor(tools): add multi-provider support to web_search tool

### PR DESCRIPTION
**Pull request by vrijmetse**
## Description

Adds Google Custom Search as an alternative provider to the existing `web_search` tool, giving users flexibility based on their existing API credentials while maintaining full backward compatibility.

**Key design decision:** Brave is preferred by default to avoid breaking existing users.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Refactoring (no functional changes)

## Related Issues

Fixes #793

## Changes Made

- Add Google Custom Search support to `web_search` tool via new `provider` parameter
- `provider="auto"` (default): Tries Brave first, falls back to Google
- `provider="brave"`: Force Brave Search API
- `provider="google"`: Force Google Custom Search API
- Register Google credentials (`GOOGLE_API_KEY`, `GOOGLE_CSE_ID`) in `credentials/search.py`
- Update README documentation for multi-provider usage
- Add comprehensive tests (13 tests covering all providers and edge cases)

## Backward Compatibility

| User Setup | Behavior |
|------------|----------|
| Has `BRAVE_SEARCH_API_KEY` only | Works exactly as before |
| Has Google keys only | Uses Google automatically |
| Has both | Uses Brave (backward compatible) |
| Wants Google when both set | Use `provider="google"` |

## Testing

```bash
cd tools && PYTHONPATH=src pytest tests/tools/test_web_search_tool.py -v
# 13 passed
```

- [x] Unit tests pass
- [x] Manual testing performed with real APIs

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Backward compatible - no breaking changes

## Screenshots (if applicable)

N/A - CLI tool, no UI changes.

### CODE REVIEW

**Comment by vrijmetse** [comment]
Let me try to resolve the conflict 🙏

File: tools/src/aden_tools/tools/web_search_tool/README.md [hunk 1]
```diff
@@ -1,31 +1,64 @@
...
+| `num_results` | int | No | `10` | Number of results (1-10 for Google, 1-20 for Brave) |
+| `country` | str | No | `us` | Country code for localized results |
+| `language` | str | No | `en` | Language code (Google only) |
+| `provider` | str | No | `auto` | Provider: "auto", "google", or "brave" |
 
 ## Environment Variables
 
+Set credentials for at least one provider:
+
+### Option 1: Google Custom Search (Recommended)
```
Comment (Hundao) [inline tools/src/aden_tools/tools/web_search_tool/README.md]: "Recommended" here conflicts with auto mode defaulting to Brave first. Suggest removing "(Recommended)".

**Review by Hundao — COMMENTED** [review]
Clean implementation. Good separation between providers and backward compatible. Two things:

1. PR description checks "Breaking change" but this is backward compatible — please uncheck that box.
2. There are merge conflicts — please rebase on latest main and resolve.

File: tools/src/aden_tools/tools/web_search_tool/README.md (hunk 1 above)
Comment (vrijmetse) [reply tools/src/aden_tools/tools/web_search_tool/README.md]: Removed as suggested
